### ISSUE TITLE
Lua function runtime error with HAProxy 2.8

### ISSUE THREAD

**Issue by nightah** [TimWolla/haproxy-auth-request#58]
It seems during some routine dependency bumps that our integration testing for Authelia has run into issues with HAProxy 2.8.

I initially thought this might be due to our CI environment or an older version of `haproxy-auth-request` or `haproxy-lua-http`.
I attempted to manually re-run the same tests locally in the following scenarios:
1. Repo versions of the aforementioned lua scripts (https://github.com/authelia/authelia/tree/renovate/haproxy-2.x/internal/suites/example/compose/haproxy)
2. Updated `haproxy-lua-http` (https://github.com/haproxytech/haproxy-lua-http/tree/4ac4483142c4f12445d47e55c8ecf155d9be0904)

The errors in both scenarios seem to match our CI/build environment errors:

```
Lua function 'auth-intercept': [state-id 0] runtime error: table index is nil from [C]: in method 'get_stats', /usr/local/etc/haproxy/auth-request.lua:136: in global 'auth_request', /usr/local/etc/haproxy/auth-request.lua:35: in function line 31. 
[ALERT]    (3) : Lua function 'auth-intercept': [state-id 0] runtime error: table index is nil from [C]: in method 'get_stats', /usr/local/etc/haproxy/auth-request.lua:136: in global 'auth_request', /usr/local/etc/haproxy/auth-request.lua:35: in function line 31.
```

You can also see full log output at our [CI build](https://buildkite.com/authelia/authelia/builds/21098#01887383-cc88-4797-98dc-eb8ba004f4ca)

The HAProxy configuration can also be found [here](https://github.com/authelia/authelia/blob/renovate/haproxy-2.x/internal/suites/example/compose/haproxy/haproxy.cfg), rolling back to HAProxy 2.7.8 results in a working environment.

**Comment by TimWolla** [comment]
Thanks for the report. Apparently the CI config in this repo is messed up and does not properly report a failure:

https://github.com/TimWolla/haproxy-auth-request/actions/runs/5101534441/attempts/1#summary-13811454437

:sob:

**Comment by TimWolla** [comment]
CI is fixed now, but the issue is not yet.

**Comment by nightah** [comment]
Thanks for the update @TimWolla, no rush at this stage we haven't had any reports of it affecting anybody (yet).

**Comment by TimWolla** [comment]
It's probably going to affect myself once I upgrade to HAProxy 2.8 :smile: I intentionally add the new CI early on and let it run on a weekly schedule to catch this kind of issue early – if the CI works, that is.

**Comment by TimWolla** [comment]
This will be fixed with HAProxy 2.8.1 and tests with the nightly snapshot are already green again.
